src/helpers/StatsClient.ts

```diff
@@ -0,0 +1,176 @@
+import { StatsD, Tags } from 'hot-shots';
+import winston = require('winston');
+
+export interface AddTagsToScopeOptions {
+	scope: string;
+	tags: Tags;
+}
+
+export enum PluginType {
+	ACTIVITY_ANALYZER = 'activity_analyzer',
+	AD_RENDER = 'ad_render',
+	AUDIENCE_FEED = 'audience_feed',
+	BID_OPTIMIZER = 'bid_optimizer',
+	CUSTOM_ACTION = 'custom_action',
+	EMAIL_RENDERER = 'email_renderer',
+	EMAIL_ROUTER = 'email_router',
+	RECOMMENDER = 'recommender',
+}
+
+export enum METRICS_NAME {
+	EXTERNAL_API_CALL_SUCCESS = 'external_api_call_success',
+	EXTERNAL_API_CALL_ERROR = 'external_api_call_error',
+	EXTERNAL_API_CALL_RETRY = 'external_api_call_retry',
+
+	MICS_API_CALL_SUCCESS = 'mics_api_call_success',
+	MICS_API_API_CALL_ERROR = 'mics_api_call_error',
+	MICS_API_API_CALL_RETRY = 'mics_api_call_retry',
+
+	AUDIENCE_FEED_BATCH_UPSERT = 'audience_feed_batch_upsert',
+	AUDIENCE_FEED_BATCH_DELETE = 'audience_feed_batch_delete',
+
+	AD_RENDER_BANNER_GENERATED_DEFAULT = 'ad_render_banner_generated_default',
+	AD_RENDER_BANNER_GENERATED_CUSTOM = 'ad_render_banner_generated_custom',
+
+	ACTIVITY_CREATED = 'activity_created',
+	ACTIVITY_FAILED_TO_CREATE = 'activity_failed_to_create',
+}
+
+export interface InitOptions {
+	/**
+	 * interval to send stats to datadog in ms (default = 10 minutes)
+	 */
+	timerInMs?: number;
+
+	/**
+	 * default plugins tags
+	 */
+	tags: Tags & { pluginType: PluginType; pluginName: string };
```

Comment:
The artifact id is already pushed as `plugin_artifact_id` by the datadog integration, do we need an extra `pluginName`?
(we can easily add the plugin type in the datadog integration too)

Reply:
Ok I remove both PluginType and PluginName as you will deal with them on your side